### ISSUE TITLE
Multiple QueryParams aren't delimited with &

### ISSUE THREAD

**Issue by declension** [haskell-servant/servant-quickcheck#23]
I may have misunderstood, so am quoting the test api / spec I used to narrow down the problem:

### Test API
With this basic API:
```haskell
type QueryStringAPI = "path" :> QueryParam "one" SafeText :> QueryParam "two" SafeText :> Get '[JSON] Text
qsApi:: Proxy QueryStringAPI
qsApi = Proxy

newtype SafeText = SafeText Text
instance ToHttpApiData SafeText
  where toUrlPiece (SafeText st) = st
```

### Spec
..and a `serverSatisifies` spec like:
```haskell
burl = BaseUrl Http "localhost" 8080 ""
spec = it "Query String API doesn't return 500" $
      serverSatisfies qsApi burl stdArgs
        (not500 <%> mempty)

instance Arbitrary SafeText
    -- Ensure it's not the random characters being the problem
    where arbitrary = return $ SafeText "value"
```

### Output
Using a basic HTTP server (`python -m SimpleHTTPServer 8080` here), I see lots of this in the logs:

<tt>127.0.0.1 - - [24/Feb/2017 18:01:06] "GET /path/<strong>?two=valueone=value</strong> HTTP/1.1" 404 -</tt>

Whilst I think the `path/` is covered separately in #22, it's the missing `&` here that means I can't query the real API I was examining. Is this a bug?

Note: a client created with `servant-client` works as expected, with ampersands delimiting multiple (separate) `QueryParam`s...

**Comment by declension** [comment]
(for reference, from IRC): [Line 95](https://github.com/haskell-servant/servant-quickcheck/blob/master/src/Servant/QuickCheck/Internal/HasGenRequest.hs#L95) (probably) needs to check whether the queryString is empty, and if it's not, use '&' before adding the param

**Comment by declension** [comment]
(Also realised this applies to `QueryFlag`s)

**Comment by jkarni** [comment]
Released as 0.0.2.3
